Xwt.Gtk/Xwt.GtkBackend/TableViewBackend.cs

```diff
@@ -514,6 +519,40 @@ static CustomTreeView ()
 				GtkWorkarounds.RemoveKeyBindingFromClass (Gtk.TreeView.GType, Gdk.Key.BackSpace, Gdk.ModifierType.None);
 		}
 
+		protected override bool OnButtonPressEvent (EventButton evnt)
+		{
+			// If we are clicking on already selected row, delay the selection until we are certain that
+			// the user is not starting a DragDrop operation. 
+			// This is needed to allow user to drag multiple selected rows.
+			TreePath treePath;
+			TreeViewColumn column;
+			GetPathAtPos ((int)evnt.X, (int)evnt.Y, out treePath, out column);
+			var ctrlShiftMask = (evnt.State & (Gdk.ModifierType.ShiftMask | Gdk.ModifierType.ControlMask | Gdk.ModifierType.Mod2Mask));
+			if (evnt.Button == 1 && this.Selection.PathIsSelected (treePath) && ctrlShiftMask == 0) {
+				delayedSelection = treePath;
+				delayedSelectionColumn = column;
+				delayedSelectionX = evnt.X;
+				delayedSelectionY = evnt.Y;
+				Selection.SelectFunction = (_, __, ___, ____) => false;
+				var result = base.OnButtonPressEvent (evnt);
+				Selection.SelectFunction = (_, __, ___, ____) => true;
+				return result;
+			}
+			return base.OnButtonPressEvent (evnt);
+		}
+
+		protected override bool OnButtonReleaseEvent (EventButton evnt)
+		{
+			// Now, if mouse hadn't moved, we are certain that this was just a click. Proceed as usual.
+			if (delayedSelection != null) {
+				if (evnt.X.Equals (delayedSelectionX) && evnt.Y.Equals (delayedSelectionY))
```

Comment:
Checking for exact equality seems a bit rigid and we are making assumptions about what constitutes a drag. How about we remove `delayedSelectionX` and `delayedSelectionY` and instead set `delayedSelection` to null in the drag started event?

Reply:
The problem with this is that I can't seem to find propert drag started or drag motion event. I've tried overriding
```
protected override bool OnDragMotion(Gdk.DragContext context, int x, int y, uint time)
{
	return true;
}
```
but it doesn't get called. 😓

Reply:
Nevermind, I've found a way :)